### PR TITLE
Fix don't show inserter in Zoom Out dropzone when the text is visible

### LINKED ISSUE #67568: Don't show inserter in Zoom Out dropzone when the text is visible
### Description

When clicking the `+` inserter in between sections in Zoom Out the inserter `+` remains visible and thus obscures the text.

The inserter should hide once the dropzone is activated.

### Step-by-step reproduction instructions

1. Zoom Out
2. Click inserter in between sections
3. See text obscured.

### Screenshots, screen recording, code snippet

Image

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

### PR DESCRIPTION
**Pull request by getdave**
## What?

Hides the inserter `+` when the zoom out dropzone is rendered

Closes https://github.com/WordPress/gutenberg/issues/67568

## Why?

Because otherwise - as per the Issue - we get the inserter rendered on top of the drop zone text in Zoom Out.

## How?

Conditionally displayed the inserter only when the current insertion index doesn't match the targetted insertion index for the inserter being rendered.

## Testing Instructions

- Site Editor
- Add Patterns
- Zoom Out
- Select a section
- Click on the `+` inserter
- See that the dropzone appears 
- See the inserter doesn't render
- Click another section and its inserter `+`
- Check that works too
- Generally try and break it - use 0 blocks, try Pages, Templates...etc

## Screenshots or screencast 

https://github.com/user-attachments/assets/6057b4fb-4938-46cb-9bb2-9ef1ac745e50

### CODE REVIEW

File: packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js [hunk 1]
```diff
@@ -62,7 +68,19 @@ function ZoomOutModeInserters() {
...
 		( clientId ) => selectedBlockClientId === clientId
 	);
-	const nextClientId = blockOrder[ index + 1 ];
+
+	const insertionIndex = index + 1;
+
+	const nextClientId = blockOrder[ insertionIndex ];
+
+	// if the block insertion point is visible, and the insertion
+	// indicies match then we don't need to render the inserter.
```
Comment (MaggieCabrera) [inline packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js:77]:
```suggestion
	// indices match then we don't need to render the inserter.
```
Super nit

**Review by MaggieCabrera — APPROVED** [review]
This LGTM, tests ok and the code makes sense

File: packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js [hunk 2]
```diff
@@ -62,7 +68,19 @@ function ZoomOutModeInserters() {
 	const index = blockOrder.findIndex(
 		( clientId ) => selectedBlockClientId === clientId
 	);
-	const nextClientId = blockOrder[ index + 1 ];
+
+	const insertionIndex = index + 1;
+
+	const nextClientId = blockOrder[ insertionIndex ];
+
+	// if the block insertion point is visible, and the insertion
```
Comment (mikachan) [inline packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js]:
```suggestion
	// If the block insertion point is visible, and the insertion
```

**Review by mikachan — APPROVED** [review]
Also LGTM!

File: packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js (hunk 1 above)
Comment (getdave) [reply packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js:77]: Missed this one. Apologies

Comment (getdave) [reply packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js:77]: @MaggieCabrera Please can you approve https://github.com/WordPress/gutenberg/pull/68051

**Pull request merged**